infra/auth-server/src/utils/verify-sign.js

```diff
@@ -0,0 +1,52 @@
+'use strict'
+
+const Web3Utils = require('web3-utils')
+
+const Eth = require('web3-eth')
+const Web3Eth = new Eth()
+
+const stringify = require('json-stable-stringify')
+
+const logger = require('../logger')
+
+/**
+ * Verifies the sign of message
+ *
+ * @param {String} address ETH address of the signer
+ * @param {String} signature The actual signature
+ * @param {Object} payload The data that was signed
+ * @returns {Boolean} `true` if `signature` is signed by `address`
+ *                    on `payload` in the last 15 minutes;
+ *                    `false` otherwise
+ */
+const verifySign = ({ address, signature, payload }) => {
+  try {
+    const now = Date.now()
+    const signExpLimit =
+      (parseInt(process.env.SIGN_EXPIRES_IN) || 15) * 1000 * 60
+    const minTimestamp = now - signExpLimit
+
+    if (payload.timestamp < minTimestamp || payload.timestamp > now) {
```

Comment:
Because of client side clock differences, we should probably also allow perhaps one minute or five minutes into the future.

Reply:
Makes sense. Added.